links/link.go

```diff
@@ -301,26 +366,38 @@ const (
 // and return a function that can run in the netns.Do() call for execution in a network namespace.
 func SetNameMACAndUpInterface(l netlink.Link, endpt Endpoint) func(ns.NetNS) error {
 	return func(_ ns.NetNS) error {
-		// rename the given link
-		err := netlink.LinkSetName(l, endpt.GetIfaceName())
-		if err != nil {
-			return fmt.Errorf(
-				"failed to rename link: %v", err)
+		// rename the link created with random name if its length is acceptable by linux
+		if len(endpt.GetIfaceName()) < 15 {
+			err := netlink.LinkSetName(l, endpt.GetIfaceName())
+			if err != nil {
+				return fmt.Errorf(
+					"failed to rename link: %v", err)
+			}
+		} else {
+			// else we set the desired long name as alias
+			// in future we need to set it as an alternative name,
+			// pending https://twitter.com/ntdvps/status/1709580216648024296
+			err := netlink.LinkSetAlias(l, endpt.GetIfaceName())
+			if err != nil {
+				return fmt.Errorf(
+					"failed to add alias: %v", err)
+			}
```

Comment:
Here is a ~big change I made to the original proposal.
Instead of creating some hashed name for the interface I propose the following:

1. if an interface name is 14+ chars we use the random name that has already been generated by clab and use it
2. the long name we set as an alias for the interface. The alias is like a description of the interface that can container 128 chars. The alias though doesn't let you use `ip link show <alias>`.
3. I am waiting on this to merge https://github.com/vishvananda/netlink/pull/862 to use alternative name instead of the alias. The altname allows users to use it in `ip` utility.